cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java

```diff
@@ -737,22 +742,50 @@ public QueryHandle createNamespace(Id.Namespace namespace) throws ExploreExcepti
       SessionHandle sessionHandle = null;
       OperationHandle operationHandle = null;
 
+      // Get custom Hive database name if one was  provided
+      customHiveDatabase = getCustomHiveDatabaseIfExists(namespace.getId());
       try {
         sessionHandle = cliService.openSession("", "", sessionConf);
-
-        String database = getHiveDatabase(namespace.getId());
-        // "IF NOT EXISTS" so that this operation is idempotent.
-        String statement = String.format("CREATE DATABASE IF NOT EXISTS %s", database);
-        operationHandle = doExecute(sessionHandle, statement);
-        QueryHandle handle = saveReadOnlyOperation(operationHandle, sessionHandle, sessionConf, statement, database);
-        LOG.info("Creating database {} with handle {}", namespace, handle);
+        QueryHandle handle;
+        if (Strings.isNullOrEmpty(customHiveDatabase)) {
```

Comment:
I don't understand this logic. We only try and create the database if a custom one is not set?  If somebody sets a custom hive database they need to create it themselves before creating the cdap namespace?  Who will know to do that? Will any permissions errors be correctly propagated back up to the user?